qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -295,10 +296,15 @@ def _format_data(
                     continue
                 if len(g_values) == 1:
                     averaged[k] = v[0]
+                elif all(isinstance(v, numbers.Number) for v in g_values):
```

Comment:
I think a different variable should be used here to avoid confusion with the `v` in the larger loop.